compiler/rustc_mir_build/src/thir/pattern/mod.rs

```diff
@@ -468,11 +469,9 @@ impl<'a, 'tcx> PatCtxt<'a, 'tcx> {
         let instance = match ty::Instance::resolve(self.tcx, param_env_reveal_all, def_id, substs) {
             Ok(Some(i)) => i,
             Ok(None) => {
-                self.errors.push(if is_associated_const {
-                    PatternError::AssocConstInPattern(span)
-                } else {
-                    PatternError::StaticInPattern(span)
-                });
+                if is_associated_const {
```

Comment:
I think you can assert that this bool is true, we should always report an error here, as otherwise I think we are miscompiling

Reply:
Added a debug assertion with a comment, is that correct?